private/pkg/git/git.go

```diff
@@ -47,13 +46,6 @@ const (
 	ModeSubmodule ObjectMode = 016_0000
 )
 
-const (
-	// refTypeHash is the reference type for a git SHA hash.
-	refTypeHash referenceType = "hash"
-	// refTypeBranch is the reference type for a git branch.
-	refTypeBranch referenceType = "branch"
-)
```

Comment:
Converted to structs implementing the same interface, because the branch reference has an optional remote name.